homework/kaydunov_denis/stream/src/main/java/github/kaydunovdenis/service/LibraryService.java

```diff
@@ -0,0 +1,129 @@
+package github.kaydunovdenis.service;
+
+import github.kaydunovdenis.bean.Author;
+import github.kaydunovdenis.bean.Book;
+import lombok.extern.log4j.Log4j;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Log4j
+public class LibraryService {
+    private final Map<String, Book> books;
+
+    public LibraryService(Map<String, Book> books) {
+        this.books = books;
+    }
+
+    /**
+     * @param countPages of book
+     * @return I. find Books With Pages More Than countPages
+     */
+    public List<Book> findBooksWithPagesMoreThan(int countPages) {
+        return books.values().stream()
+                .filter(book -> book.getNumberOfPages() > countPages)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * @return II. find the books with max number of pages;
+     */
+    public List<Book> findBooksWithMaxCountPages() {
+        int maxPages = findMaxCountPagesAmongAllBooks();
+        return books.values().stream()
+                .filter(book -> book.getNumberOfPages() == maxPages)
+                .collect(Collectors.toList());
+    }
+
+    private int findMaxCountPagesAmongAllBooks() {
+        return books.values().stream()
+                .map(Book::getNumberOfPages)
+                .max(Integer::compareTo)
+                .orElse(0);
+    }
+
+    /**
+     * @return II. find the books with min number of pages;
+     */
+    public List<Book> findBooksWithMinCountPages() {
+        int minPages = findMinCountPagesAmongAllBooks();
```

Comment:
U can use it in method without creating local variable

Reply:
How can I do this without creating a variable?
This variable is used several times.
This means that each time its value will be calculated over and over again.

Reply:
@DenisAbramovDm 
How can I do this without creating a variable?
This variable is used several times.
This means that each time its value will be calculated over and over again.

Reply:
As I can see you are using minPages variable only once in this method

Reply:
This is not true. Inside the stream variable "minPages" is using many times during comparison
book.getNumberOfPages() == minPages